File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        
        vector<int> v;
        
        for (int i = 0; i<nums.size(); i++){
            int newIndex = abs(nums[i])-1;
            if (nums[newIndex] > 0) nums[newIndex] *= -1;
        }
        for (int i = 1; i<=nums.size(); i++){
            if (nums[i-1] > 0) v.push_back(i);
        }
        return v; 
    }
};

File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        
        List<Integer> missingIntegers = new LinkedList<Integer>();

        
        for (int i = 0; i < nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            nums[index] = -Math.abs(nums[index]);
        }
        
        

        for (int i = 1; i <= nums.length; i++) {
            if (nums[i - 1] > 0) missingIntegers.add(i);
        }
        
        return missingIntegers;
    }
}